### ISSUE TITLE
How to support multi-channels(>3) images?

### ISSUE THREAD

**Issue by Cogito2012** [ShaoqingRen/faster_rcnn#87]
I had prepared my 4 channels tiff image datasets, and modified all the .m files where errors occured, and faster_rcnn can successfully run a result, but mAP remained the same as original RGB images datasets configuration. So I think whether it's because I didn't modify train_val.prototxt and test.prototxt files' data layer:
=======================
input: "data"
input_dim: 1
input_dim: 3
input_dim: 224
input_dim: 224
========================
 
But if I changed all prototxts' input_dim: 3 as input_dim: 4, errors will occured in functions/rpn/proposal_train.m , the line
caffe_solver.net.copy_from(opts.net_file);

So anyone know how to change faster_rcnn to support multi-channels images input?

**Comment by CHUNYUWANG** [comment]
When you change the prototxt  (input_dim: 3 as input_dim: 4), it changes the number of parameters for the first convolution layer. So you can't load the original weights. I think you can change the "name" of the first convolution layer so that caffe ignores the weights of that layer.

**Comment by xiaoxiongli** [comment]
@CHUNYUWANG  yes, in caffe, if you do not want copy the weight for a special layer, you need rename the layer.
